metrics/storage/fio_job.sh

```diff
@@ -72,24 +77,49 @@ EOF
 
 # Create the job configuration for FIO
 # this job will run inside of the container.
-function create_fio_job()
+# This job sets up the image - creates the test file
+function create_fio_prep_job()
 {
 	FIO_JOB="$(cat << EOF
 	#!/bin/bash
 	# This fio job will run inside of
 	# the container.
+	# It only runs for 1s, its job is to create
+	# the test file
 
 	fio --ioengine="$ENGINE" \
 		--name="$FIO_TEST_NAME" \
 		--filename="$FILE_NAME" \
 		--bs="$BLOCK_SIZE" \
 		--size="$FILE_SIZE" \
 		--readwrite="$OPERATION" --max-jobs=$NUM_JOBS \
+		--runtime=1 \
 		${EXTRA_ARGS}
 EOF
 )"
 }
 
+# Create the job configuration for FIO
+# this job will run inside of the container.
+# This job runs the actual test
+function create_fio_run_job()
+{
+	FIO_JOB="$(cat << EOF
+	#!/bin/bash
+	# This fio job will run inside of
+	# the container.
+
+	fio --ioengine="$ENGINE" \
```

Comment:
We might not care (and I see there are already existing functions similar to this), but you could make that `exec fio` to avoid creating a 2nd process. In fact, why don't we just call `fio` directly? I've just noticed that `$FIO_JOB` itself is called via `bash -c` so we have 3 processes I think?

Reply:
Heh - we probably don't care much, but I took out the `#!/bin/bash`, and the world seems fine...
I've not injected the `exec`, as I've not seen that as a common use case on a docker command line.
Mostly I'd like to keep the script readable - I think that is how the original layout came about for those commands.